### PR TITLE
[AndroidManifest] Add support for `<property>` element via `PropertyAttribute`.

### PR DESCRIPTION
**Pull request by jpobst**
Fixes: https://github.com/xamarin/xamarin-android/issues/8729

Adds support for specifying the `AndroidManifest.xml` [`<property>`](https://developer.android.com/guide/topics/manifest/property-element) element via a new `[Property]` attribute.  This appears to be the exact same as `<meta-data>` but with a different name, so it is implemented exactly the same.

Example:

```csharp
[Service (Name = "fooService", ForegroundServiceType = Android.Content.PM.ForegroundService.TypeSpecialUse)]
[Property ("android.app.PROPERTY_SPECIAL_USE_FGS_SUBTYPE", Value = "explanation_for_special_use")]
public class FooService : Service
{ ... }
```

creates:

```xml
<service android:name="fooService" android:foregroundServiceType="specialUse">
  <property android:name="android.app.PROPERTY_SPECIAL_USE_FGS_SUBTYPE"
      android:value="explanation_for_special_use" />
</service>
```

This element can be applied to `<application>`, `<activity>`, `<provider>`, `<receiver>`, and `<service>` elements.

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ManifestTest.cs [hunk 1]
```diff
@@ -1211,5 +1212,122 @@ public void IntentFilterDataPathsTest ()
...
 	}
 }
+
+// Dummy types because we don't want to take a dependency on Mono.Android or Java.Interop
+namespace Android.App
+{
+	public class Activity : Java.Interop.IJavaPeerable { }
+	public class Application : Java.Interop.IJavaPeerable { }
+	public class Service : Java.Interop.IJavaPeerable { }
+}
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ManifestTest.cs:1322]: Should we put these in a separate file, to make them easier to find? I could see another test wanting to use these at some point, and they are useable within the entire project.

File: src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs [hunk 2]
```diff
@@ -579,6 +580,10 @@ XElement CreateApplicationElement (XElement manifest, string applicationClass, L
 				Assemblies.SelectMany (path => MetaDataAttribute.FromCustomAttributeProvider (Resolver.GetAssembly (path), cache))
 					.Where (attr => attr != null)
 					.ToList ();
+			var properties =
+				Assemblies.SelectMany (path => PropertyAttribute.FromCustomAttributeProvider (Resolver.GetAssembly (path), cache))
+					.Where (attr => attr != null)
+					.ToList ();
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs:586]: This shouldn't be fixed here, but I wonder if the N `Assemblies.SelectMany()` calls could be a performance issue.

**Comment by jonpryor** [comment]
Proposed commit message:

```markdown
Fixes: https://github.com/xamarin/xamarin-android/issues/8729

API-31 added support for a [`<property/>`][0] element within
`AndroidManifest.xml`, which can be contained within `<activity/>`, 
`<application/>`, `<provider/>`, `<receiver/>`, and `<service/>`
elements.

Add a new `Android.App.PropertyAttribute` custom attribute which will
emit `<property/>` within `AndroidManifest.xml`:

	[AttributeUsage(AttributeTargets.Assembly | AttributeTargets.Class,
	        AllowMultiple = true, Inherited = false)]
	public partial class PropertyAttribute : Attribute {

	    public string Name      {get;}
	    public string? Resource {get;}
	    public string? Value    {get;}

	    public PropertyAttribute (string name);
	}

Semantics otherwise appear to be identical to
`<meta-data/>`/`MetaDataAttribute`: only `Resource` or `Value` can
be specified, etc.  (This is checked by `aapt2`.)

Using `[assembly:Property(…)]` will result in `<property/>` being
emitted within the `<application/>` element.

For example:

	[Service (Name = "fooService", ForegroundServiceType = Android.Content.PM.ForegroundService.TypeSpecialUse)]
	[Property ("android.app.PROPERTY_SPECIAL_USE_FGS_SUBTYPE", Value = "explanation_for_special_use")]
	public partial class FooService : Service {
	}

will emit the `AndroidManifest.xml` fragment:

	<service android:name="fooService" android:foregroundServiceType="specialUse">
	  <property
	      android:name="android.app.PROPERTY_SPECIAL_USE_FGS_SUBTYPE"
	      android:value="explanation_for_special_use"
	  />
	</service>

[0]: https://developer.android.com/guide/topics/manifest/property-element
```

**Comment by entdark** [comment]
Hi, what version will it appear in?

**Comment by jpobst** [comment]
It will be in .NET 9. Preview 7.

**Pull request merged**